pkg/reporting/cookbooks.go

```diff
@@ -101,87 +108,72 @@ func (r *CookbookRecord) NumCorrectable() int {
 	return i
 }
 
-func NewCookbooks(
-	cbi CookbookInterface, searcher SearchInterface,
-	runCookstyle, onlyUnused bool, workers int,
-) (*CookbooksStatus, error) {
+func NewCookbooks(cbi CookbookInterface, searcher SearchInterface, runCookstyle bool, onlyUnused bool, workers int) (*CookbooksStatus, error) {
```

Comment:
We talked about this split in a different context - I've separated out NewCookbooks from Run so that we can return data that the caller needs (total count, progress channel) and have it available before the execution begins.